plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/perspective/widgets/machine/appliance/recipe/RecipeTabPresenter.java

```diff
@@ -43,23 +43,28 @@ public RecipeTabPresenter(RecipeView view, RecipeScriptDownloadServiceClient rec
 
     /**
      * Calls special method on view which updates recipe of current machine.
+     * If recipe type is dockerfile, then it's content can be fetched.
      *
      * @param machine
      *         machine for which need update information
      */
     public void updateInfo(@NotNull final Machine machine) {
-        recipeScriptClient.getRecipeScript(machine).then(new Operation<String>() {
-                    @Override
-                    public void apply(String recipe) throws OperationException {
-                        view.setScript(recipe);
-                    }
-                }).catchError(new Operation<PromiseError>() {
-            @Override
-            public void apply(PromiseError error) throws OperationException {
-                Log.error(RecipeTabPresenter.class,
-                          "Failed to get recipe script for machine " + machine.getId() + ": " + error.getMessage());
-            }
-        });
+        if (machine.getRecipeType().equals("dockerfile")) {
+            recipeScriptClient.getRecipeScript(machine).then(new Operation<String>() {
+                @Override
+                public void apply(String recipe) throws OperationException {
+                    view.setScript(recipe);
+                }
+            }).catchError(new Operation<PromiseError>() {
+                @Override
+                public void apply(PromiseError error) throws OperationException {
+                    Log.error(RecipeTabPresenter.class,
+                              "Failed to get recipe script for machine " + machine.getId() + ": " + error.getMessage());
+                }
+            });
+        }
+        view.setScript("");
```

Comment:
@mkuznyetsov if the recipe type is "image", we should still show the name of the image that we pulled.  The "image", in this case, is still the definition of the recipe.  You could say something like, "no user-provided recipe, pulled image "codenvy/ubuntu_jdk8".